### ISSUE TITLE
The `tmt link` command doesn't work well 

### ISSUE THREAD

**Issue by skycastlelily** [teemtee/tmt#3385]
Try to add link by running:
`tmt link names beaker --link https://issues.redhat.com `
nothing happens.
It looks like there is at least one issue:
1.we need to change [this line](https://github.com/teemtee/tmt/blob/main/tmt/utils/jira.py#L203) to 
`for link in links.get("relates"): `
because  links are  tmt.base.Links whose default  relation is "relates"
2.After I made that change, and run the command,it looks like work well,but after I look into my plan,I found the comment in 
the plan is removed:

```
summary: Basic smoke test
provision:
    how: beaker
    kickstart:
        script: |
            lang en_US.UTF-8
            zerombr
            clearpart --all --initlabel
            part /boot --fstype="ext4" --size=200
            part / --fstype="btrfs" --size=10000 --grow
        metadata:
            # we need this
            no_autopart
execute:
  - script: touch /tmp/lili
```
vs 

```
summary: Basic smoke test
provision:
    how: beaker
    kickstart:
        script: |
            lang en_US.UTF-8
            zerombr
            clearpart --all --initlabel
            part /boot --fstype="ext4" --size=200
            part / --fstype="btrfs" --size=10000 --grow
        metadata: no_autopart
execute:
  - script: touch /tmp/lili
link:
  - relates: https://issues.redhat.com
```
I'm not sure this is wanted behavior,IMO,it is very common that we leave some comments in testcases, for others' convenience,
and we may also forget why a specific line is needed.

Just to be clear, this issue has nothing to do with  https://github.com/teemtee/tmt/pull/3339
 It's just that I found this while testing that mr, and fvagner-config behaves the same(last run, link) as main branch,just as its comment claims:)

**Comment by psss** [comment]
Yes, currently only the `verifies` relation was used for linking. Other relations should be supported as well. When adding the link to Jira, however, we should make sure the right Jira relation si stored as well.

Regarding the removed comment, I'm afraid this is related to the way how `ruamel.yaml` works. I think it tries to store as much of the format as possible, not sure we will be able to preserve everything.
